lib/src/working_copy.rs

```diff
@@ -108,6 +108,9 @@ pub trait LockedWorkingCopy {
     /// Update to another tree without touching the files in the working copy.
     fn reset(&mut self, new_tree: &MergedTree) -> Result<(), ResetError>;
 
+    /// Update to the empty tree without touching the files in the working copy.
+    fn reset_to_empty(&mut self) -> Result<(), ResetError>;
```

Comment:
Maybe `reset(new_tree)` could fall back to this mode if `old_tree` was missing.

@martinvonz any thoughts?

Reply:
That would probably also be fine. I kind of like being more explicit like this separate method since we don't expect the tree to ever be missing unless the operation is also missing. A missing tree would indicate a corrupt repo, which might be good to tell the user about. But I don't feel strongly either way.

Reply:
> I kind of like being more explicit like this separate method since we don't expect the tree to ever be missing unless the operation is also missing. A missing tree would indicate a corrupt repo, which might be good to tell the user about.

Yeah, that's a good point. OTOH, it's a bit weird that `reset_to_empty()` doesn't fail but `reset(empty_tree)` does. It might be a problem of function naming.

Reply:
I don't think `reset()` should do any falling back - it's called during normal operation whenever we check out, so if something unexpected happens, the user should know. Regarding naming, I purposely chose something that sounds like `reset()` so that implementers would know that both these functions need to do similar things, but if you can think of a better name, we can discuss it.